File: server/Models/Account.cs
using server.Models;

namespace instaCult.Models;


public class Profile : RepoItem<string>{
  public string Name { get; set; }
  public string Picture { get; set; }

}

public class Account : Profile
{
  public string Email { get; set; }
  public int CreditCardNumber { get; set; }
}

public class Cultist : Profile{
  public int CultMemberId { get; set; }
  public int CultId {get; set;}
}

File: client/src/models/Account.js
export class Profile{
  constructor(data){
    this.id = data.id
    this.name = data.name
    this.picture = data.picture
  }
}

export class Account extends Profile {
  constructor(data) {
    super(data)
    this.email = data.email
    // TODO add additional properties if needed
  }
}

export class Cultist extends Profile{
  constructor(data){
    super(data)
      this.cultMemberId = data.cultMemberId
  }
}
